packages/intrepid2/src/Cell/Intrepid2_CellToolsDefControlVolume.hpp

```diff
@@ -386,6 +386,7 @@ namespace Intrepid2 {
     INTREPID2_TEST_FOR_EXCEPTION( cellCoords.extent(2) != primaryCell.getDimension(), std::invalid_argument,
                                   ">>> ERROR (Intrepid2::CellTools::getSubcvCoords): cell coords dimension(2) does not match to the dimension of the cell." );
 #endif
+    using deviceType = typename decltype(subcvCoords)::device_type;
```

Comment:
@mperego, I know you and @kyungjoo-kim are already discussing this.  I do think that we need to get some clarity around the meaning of the CellTools DeviceType template argument and what we require of the arguments.  My feeling is that the rule should be that all view arguments should have memory spaces that are accessible from DeviceType::memory_space.  We could add `static_assert`s to that effect.

Reply:
(If you agree with that, I would drop this typedef, and just use `DeviceType` throughout.

Reply:
Yes, I can do that, as long as we agree on it. It might be good to go over this at our meeting on Monday.

Reply:
deviceType sounds like a typo of DeviceType. Someone may confuse. what about saying local_device_type ? or device_type ?